### DISCUSSION TITLE
Plan to generate MSFT_lod

### DISCUSSION THREAD

**Discussion by arpu** [zeux/meshoptimizer#256] [Ideas]
Hey,

do you have any Plans to implement MSFT_lod? looks like Three gltf Loader  will get this with 
https://github.com/takahirox/three-gltf-extensions/blob/main/loaders/MSFT_lod/MSFT_lod.js

**Comment by zeux** [comment]
No current plans. I'm focusing on core algorithms for the moment - this would generally be within gltfpack's scope (and could dovetail with the existing simplification support), but it would be nice to see a bit more official support for LODs - for example:

- A more official extension (not MSFT-prefixed), potentially in a slightly different form so that duplicating nodes isn't critical
- Native support in Three.js/Babylon.js (the link you provided is an extension to three.js)

Still, I could eventually implement support for MSFT_lod if by the time I am done with all of the core simplification algorithms and feel like they are fully complete MSFT_lod is still the status quo. Or I could help define a cross-vendor extension - time will tell. So this is a good idea, just not on my radar in the immediate future.

**Comment by arpu** [comment]
There is some new https://github.com/KhronosGroup/glTF/pull/2228
